File: data_structures/stack.js
//Create a stack using pseudoclassical instantiation
var Stack = function(limit) {
  this._storage = [];
  this._size = 0;
  this.limit = limit || 1000;
}

Stack.prototype.push = function(val) {
  //Increment size and add value to top of the stack
  if (this._size < this.limit) {
    this._size++;
    this._storage.push(val);
  } else {
    return "Could not add value, stack is full";
  }
}

Stack.prototype.pop = function(val) {
  //Remove and return top item if stack is non-empty
  if (this._size > 0) {
    this._size--;
    return this._storage.pop();
  }
  return null; 
}

Stack.prototype.size = function() {
  return this._size;
}

File: data_structures/Python/stack.py
#Create a stack class
class Stack():

	def __init__(self, limit=1000):
		self._limit = limit
		self._storage = []
		self._size = 0

	def pop(self):
		if self._size > 0:
			self._size -= 1
			return self._storage.pop()

	def add(self, item):
		if self._size < self._limit:
			self._size += 1
			self._storage.append(item)

	def size(self):
		return self._size